.github/workflows/ci.yml

```diff
@@ -279,85 +156,114 @@ jobs:
+
       - name: Install dependencies
         run: dart pub get
 
       - name: Run tests
         run: ${{ matrix.architecture == 'arm' && 'arch -arm64 ' || '' }}dart test -r expanded -j 1 --test-randomize-ordering-seed random
 
-# Flutter jobs
+      - name: Cleanup apps
+        if: always()
+        run: dart run realm_dart delete-apps --appIds '${{ steps.deploy-apps.outputs.appIds }}' --baas-url ${{ secrets.REALM_QA_URL }} --atlas-cluster ${{ needs.deploy-cluster.outputs.clusterName }} --api-key ${{ secrets.ATLAS_QA_PUBLIC_API_KEY }} --private-api-key ${{ secrets.ATLAS_QA_PRIVATE_API_KEY }} --project-id ${{ secrets.ATLAS_QA_PROJECT_ID }} --differentiator '${{ env.BAAS_DIFFERENTIATOR }}'
 
-  flutter-desktop:
-    runs-on: ${{ matrix.platform == 'linux' && 'ubuntu' || matrix.platform }}-latest
+  flutter-desktop-tests:
     strategy:
       fail-fast: false
       matrix:
-        platform: [ 'macos', 'windows', 'linux']
-    name: Flutter ${{ matrix.platform }}
+        os: ['macos', 'windows', 'linux']
+    runs-on: ${{ matrix.os == 'linux' && 'ubuntu' || matrix.os }}-latest
+    name: Flutter tests on ${{matrix.os  }}
     env:
       BAAS_CLUSTER: ${{ needs.deploy-cluster.outputs.clusterName }}
-      BAAS_DIFFERENTIATOR: flutter-${{ matrix.platform }}
     needs:
-      - baas-matrix
       - deploy-cluster
       - build-native
     steps:
+
+      - name: Generate unique job id
+        id: generate_job_id
+        if: ${{matrix.os != 'windows'}}
+        run: echo "BAAS_DIFFERENTIATOR=f$RANDOM$100" >> $GITHUB_ENV
+
+      - name: Generate unique job id on Windows
+        id: generate_job_id_win
+        if: ${{matrix.os == 'windows'}}
+        run: echo "BAAS_DIFFERENTIATOR=f$(Get-Random)" >> $env:GITHUB_ENV
+
       - name: Checkout
         uses: actions/checkout@v2
         with:
           submodules: false
 
       - name: Setup GTK
-        if: ${{ matrix.platform == 'linux' }}
+        if: ${{ matrix.os == 'linux' }}
         run: |
           sudo apt-get update -y
           sudo apt-get install -y libgtk-3-dev xvfb
 
       - name: Setup Ninja
-        if: ${{ matrix.platform == 'linux' }}
+        if: ${{ matrix.os == 'linux' }}
         uses: seanmiddleditch/gha-setup-ninja@master
 
       - name: Fetch artifacts
         uses: actions/download-artifact@v2
         with:
-          name: librealm-${{ matrix.platform }}
-          path: binary/${{ matrix.platform }}
+          name: librealm-${{ matrix.os }}
+          path: binary/${{ matrix.os }}
 
       - name: Setup Flutter
         uses: subosito/flutter-action@v2
         with:
           channel: 'stable'
 
       - name: Enable Flutter Desktop support
-        run: flutter config --enable-${{ matrix.platform }}-desktop
+        run: flutter config --enable-${{ matrix.os }}-desktop
+
+      - name: Deploy Apps for Dart tests on ${{ matrix.os }}
```

Comment:
One thing I don't like about this PR is that it moves the deploy apps step inside the test step. This means, we're adding 45-60 seconds of deploy time + a few seconds of cleanup time to each test job, when those can be done in parallel before the tests even start.

Reply:
If we move the app deployment in another jobs we will not be able to rerun the failed tests without having the app created. BTW our tests are already creating the their apps, so that if the tests are rerun they will create their apps again. So that, we can move deployment apps in a separate job and to rely that our tests are creating their apps only if they have been rerun. Just have to think how to pass app Ids to the deleting apps action.

Reply:
I repaired the workflow so that the deployments to be run in a matrix before all the tests. Unfortunately, the jobs for creating apps are taking more that the native builds time and they are somehow blocked.  For this reason it appeared that we are waiting even more. I returned deployment apps back into the tests execution job.